sdk/core/azure-core/src/base64.cpp

```diff
@@ -382,9 +382,14 @@ static void Base64WriteThreeLowOrderBytes(std::vector<uint8_t>::iterator destina
 std::vector<uint8_t> Base64Decode(const std::string& text)
 {
   auto inputSize = text.size();
+  if (inputSize > 3 && inputSize % 4 != 0)
+  {
+    throw std::runtime_error("Unexpected end of Base64 encoded string.");
+  }
+
   if (inputSize < 4)
```

Comment:
This is interesting, I am not sure we should return empty for "garbage input" like: "a"

We may want to special case only if inputSize = 0, return empty vector, otherwise, throw if input is < 4.

```suggestion
  if (inputSize == 0)
```

Reply:
We have existing tests which fail because they attempt to call Decode on "foo". I thought about updating the test, but it seemed wrong to change the test to reflect the change.

Reply:
I think "foo" falls in the category of invalid input that we should throw on. The test might have been written with the wrong assumption in mind, and imo, we should fix it.

Reply:
Empty input is valid, and divisible by 4 is valid, correct? The rest are invalid. So I also do not see a reason to not throw for `Decode("X"), Decode("XY"), or , Decode("XYZ")`.
Not sure if `"===="` or `"QQ======"` should be considered malformed.

Reply:
Ok, Ahson and Anton have pretty much convinced me. I'll push another iteration which addresses the too short input issue on Monday I think the ==== and QQ===== cases are handled because the code special cases only the last 3 characters however those are phenomenal test cases to add.